### PR TITLE
go/mio: fix maxBs calc at getOptimalBlockSz()

### PR DESCRIPTION
**Pull request by andriytk**
maxBs must be multiple of the parity group size to be optimal.

Signed-off-by: Andriy Tkachuk <andriy.tkachuk@seagate.com>

### CODE REVIEW

File: bindings/go/mio/mio.go [hunk 1]
```diff
@@ -341,25 +341,31 @@ const maxM0BufSz = 128 * 1024 * 1024
...
     }
-    usz := int(C.m0_obj_layout_id_to_unit_size(C.ulong(mio.objLid)))
     pa := &pver.pv_attr
-    gsz = usz * int(pa.pa_N)
-    /* max 2-times pool-width deep, otherwise we may get -E2BIG */
+    if pa.pa_P < pa.pa_N + 2 * pa.pa_K {
+        log.Panic("pool width (%v) is less than the parity group size" +
+                  " (%v + 2 * %v == %v), check pool parity configuration",
+                  pa.pa_P, pa.pa_N, pa.pa_K, pa.pa_N + 2 * pa.pa_K)
+    }
```
Comment (huanghua78) [inline bindings/go/mio/mio.go:356]: In LR2, we are going to handle N+k+0 configuration (No spare units).

What should the go binding do after that feature is done?

**Comment by andriytk** [comment]
I suppose in this case we will have pa_S field in the m0_pdclust_attr structure or something? If so - we could use it here then.

Any other idea / suggestion?


On 10 Jan 2021, at 16:17, Huang Hua <notifications@github.com<mailto:notifications@github.com>> wrote:



@huanghua78 commented on this pull request.

________________________________

In bindings/go/mio/mio.go<https://github.com/Seagate/cortx-motr/pull/324#discussion_r554589111>:

> +    if pa.pa_P < pa.pa_N + 2 * pa.pa_K {
+        log.Panic("pool width (%v) is less than the parity group size" +
+                  " (%v + 2 * %v == %v), check pool parity configuration",
+                  pa.pa_P, pa.pa_N, pa.pa_K, pa.pa_N + 2 * pa.pa_K)
+    }


In LR2, we are going to handle N+k+0 configuration (No spare units).

What should the go binding do after that feature is done?

File: bindings/go/mio/mio.go (hunk 1 above)
Comment (andriytk) [reply bindings/go/mio/mio.go:356]: I suppose in this case we will have pa_S field in the m0_pdclust_attr structure or something? If so - we could use it here then.

Any other idea / suggestion?

Comment (huanghua78) [reply bindings/go/mio/mio.go:356]: This task is planned. Detailed design/implementation is still in progress.

So, in future, when that is ready, this file needs to be updated.

Comment (andriytk) [reply bindings/go/mio/mio.go:356]: Yes.

**Review by huanghua78 — APPROVED** [review]

**Review by chumakd — APPROVED** [review]

**Comment by chumakd** [comment]
looks good!

**Pull request merged**